src/libcore/iter/mod.rs

```diff
@@ -520,6 +520,35 @@ impl<I> Iterator for Cycle<I> where I: Clone + Iterator {
 #[unstable(feature = "fused", issue = "35602")]
 impl<I> FusedIterator for Cycle<I> where I: Clone + Iterator {}
 
+/// An iterator that skips n elements every iteration.
+///
+/// This `struct` is created by the [`every_nth`] method on [`Iterator`]. See
+/// its documentation for more.
+///
+/// [`every_nth`]: trait.Iterator.html#method.every_nth
+/// [`Iterator`]: trait.Iterator.html
+#[must_use = "iterator adaptors are lazy and do nothing unless consumed"]
+#[stable(feature = "rust1", since = "1.0.0")]
+#[derive(Clone, Debug)]
+pub struct EveryNth<I> {
+    iter: I,
+    step: usize,
+}
+
+#[stable(feature = "rust1", since = "1.0.0")]
+impl<I> Iterator for EveryNth<I> where I: Iterator {
+    type Item = I::Item;
+
+    #[inline]
+    fn next(&mut self) -> Option<Self::Item> {
+        let elt = self.iter.next();
+        if self.step > 0 {
+            self.iter.nth(self.step - 1);
```

Comment:
> but it might also be very bad to consume elements in advance.

I wasn't clear enough, I think. I said consume "the first element" in advance, not "all." I'm not quite sure if even consuming the first element in advance could be "bad," except perhaps weakly breaking a principle.